### ISSUE TITLE
VoLTE / Wifi Calling not working with blokada

### ISSUE THREAD

**Issue by DReffects** [blokadaorg/blokada#445]
I am using a Huawei P30 Pro with Germany's O2 network.
There's no mention of anything being actually blocked by blokada within the app log, yet VoLTE and Wifi Calling is not working properly. 

### Steps to reproduce
1. Activate Blokada
2. Recieve a call
3. you cannot hear the other side + other side does not hear you

the other way around
1. Activate Blokada
2. Make an outgoing call
3. you can hear the other side but they cannot hear you

### Expected behaviour
Well... to hear the call

### Actual behaviour
Nothing can be heard

### Configuration
1. Blokada version
4.1.4

2. Device model
Huawei P30 Pro (VOG-L29)

3. Android version / ROM
Android Pie - 9.1.0.178
Android Security Patch June 5th 2019
Kernel-Version 4.9.148

### Questions
1. How long have you been using Blokada? 
~24 Hours

2. How long time ago has the problem started to happen? 
~ since the beginning

3.  How often does this problem happen to you?
every time blokada is active

**Comment by rpeter85** [comment]
I might be wrong, but it can be because the VPN is in use. Do you face the same issue with another app or service that uses the VPN API?

**Comment by DReffects** [comment]
> I might be wrong, but it can be because the VPN is in use. Do you face the same issue with another app or service that uses the VPN API?

Hey Peter, thanks for the response. I did some testing over the last few days. Wifi Calling works with VPN Services like NordVPN or even when being connected to my fritzbox vpn. The only VPN not working I found was Blokada.

**Comment by kar** [comment]
Do you have Blokada VPN subscription? Without it it's a fake "vpn" which is basically a hack and this may be the reason why it doesnt work.

**Comment by DReffects** [comment]
> Do you have Blokada VPN subscription? Without it it's a fake "vpn" which is basically a hack and this may be the reason why it doesnt work.

No, I do not have a blokada VPN Subscription, I wanted to use just the ad blocking part. Either way - I don't think this is expected behaviour.

**Comment by kar** [comment]
It might be. Android has some bugs around this. Could you try another vpn-based adblocker to see if it allows you to use the wifi calling?

**Comment by DReffects** [comment]
> It might be. Android has some bugs around this. Could you try another vpn-based adblocker to see if it allows you to use the wifi calling?

Sure, what kind of other ad blockers are there? I cannot find any in the app store.

**Comment by FrederikNS** [comment]
I'm also having problems with WiFi calls on a Huawei p30. The other end of the call receives no audio what so ever...

Interestingly I also ran Blokada on a Samsung Galaxy s10, where WiFi calls work perfectly.

Both phones are on the same carrier.

I'm not using the Blokada VPN on either phone, and settings are identical.

I did try Adguard with their free trial, and remember that they had an option in their setting to better support WiFi calls.

**Comment by DReffects** [comment]
Thanks for confirming @FrederikNS 

Intersting to read about the Galaxy S10 behaviour. What Android Versions do you run on each device?

**Comment by FrederikNS** [comment]
The Samsung Galaxy S10 is running Android 9, the version info tells me G973FXXS3ASH1/G973FOXM3ASG8/G973FXXU3ASG8

The Huawei P30 is also running Android 9, version info tells me 9.1.0.193 (C431E4R2P2)

**Comment by florianpaulhoberg** [comment]
Recently I switched to an Android device and had the same issue with a lack of fully supported Wifi calling on my Huawei on Telefonica Germany (O2) network when running Blokada. Since I'm new to Android and this device isn't rooted yet I had not that much possibilities to debug this, but afaik Wifi calling uses IPSec tunnels, so I could just suggest it may be related in something with conntrack, metric etc. When other VPNs were running fine together with Wifi calling, Blokada failed. Since other VPN solutions worked fine except of Blokada I'd suggest it wouldn't be an Android bug and took for a look for exceptions within the app. Since I'm new to Android I'm not quite sure which Service, Framework etc. relates Wifi-Calling and added some more services as probably needed, but that shouldn't make any problems for me since I'm using Blokada just for adblocking - so it would be okay to apply it only on browsers etc.. I added following exceptions for system apps and Wifi calling was working fine again with activated Blokada (keep note this are the German names):
- Android-System
- androidhwext
- Anrufverwaltung
- Call Log
- Carrier Services
- com.adnroid.carrierconfig
- com.android.cts.ctsshim
- Geräteverbindungsservice
- Geräteverbindungsservice
- Google-Deinste-Framework
- MTP-Host
- Push-Dienst
- SecureElementService
- Telefon
- Telefonservices

You may give it a try and take a look if this helps.

Regards,
Florian

**Comment by DReffects** [comment]
Heiliger Bimbam, eine Referenz auf die Bundesrepublik Deutschland :)

Servus!

Thanks for the insight and solution.

As i've upgraded to Android 10 a few days ago I've just tried blokada and found the issue has vanished. I'll do some testing and get back to you guys!

**Comment by DReffects** [comment]
Well, have it running for a few weeks now, no more issues under android 10 with a huawei p30 pro. can others confirm?
I cannot say if it's working due to an app update or the update to android 10 since i had blokada removed while it wasn't working.

**Comment by SOHEL240920** [comment]
>  The Samsung Galaxy S10 is running Android 9, the version info tells me G973FXXS3ASH1/G973FOXM3ASG8/G973FXXU3ASG8
> 
> The Huawei P30 is also running Android 9, version info tells me 9.1.0.193 (C431E4R2P2)

Same to me
